jetty-core/jetty-osgi/src/main/java/org/eclipse/jetty/osgi/OSGiDeployableBundleMetadata.java

```diff
@@ -35,27 +30,47 @@
 import org.osgi.framework.Bundle;
 import org.osgi.framework.FrameworkUtil;
 import org.osgi.framework.ServiceRegistration;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.osgi.framework.Version;
 
 /**
- * OSGiApp
- *
- * Base class representing info about a WebAppContext/ContextHandler to be deployed into jetty.
+ * Metadata useful for a deployment that will result in a {@link org.eclipse.jetty.server.handler.ContextHandler}
  */
-public class OSGiApp extends App
+public class OSGiDeployableBundleMetadata
 {
-    private static final Logger LOG = LoggerFactory.getLogger(OSGiApp.class);
+    private static final String BUNDLE = OSGiDeployableBundleMetadata.class.getPackageName() + ".bundle";
+    private static final String REGISTRATION = OSGiDeployableBundleMetadata.class.getPackageName() + ".registration";
 
-    protected Bundle _bundle;
-    protected ServiceRegistration _registration;
-    protected ContextHandler _contextHandler;
-    protected String _pathToResourceBase;
-    protected String _contextPath;
-    protected Resource _bundleResource;
+    private final Bundle bundle;
+    private final Path bundlePath;
+    private final String contextPath;
+    private final Properties properties = new Properties();
+    private String pathToResourceBase;
+
+    public OSGiDeployableBundleMetadata(Bundle bundle) throws Exception
```

Comment:
I think the `pathToResourceBase` should also be passed in to the constructor and be `final`.

Reply:
I did this originally, but the places that pathToResourceBase were being passed were in odd places in the code.
Some were late setters, some were ok to be constructors, most were blank/null/unset.

Reply:
Let's review that, I'd like to make it passed via the constructor only.

Reply:
Done.  It is now a final constructor param.